src/remediations/controller.read.js

```diff
@@ -128,6 +132,27 @@ exports.list = errors.async(async function (req, res) {
     trace.event('Fetch remediation details from DB');
     let remediations = await queries.loadDetails(req.user.tenant_org_id, req.user.username, rows);
 
+    // Check for name in fields query param:
+    // fields[data]=name
+    if (_.get(req, 'query.fields.data', []).includes('name')) {
+        trace.event('Include name data');
+        let plan_names = await queries.getPlanNames(
+            req.user.tenant_org_id
+        );
+
+        plan_names = _.map(plan_names, name => name.toJSON())
+
+        const total = fifi.getListSize(plan_names);
+        limit = limit || 1;
+        
+        trace.event('Format response');
+        const resp = format.planNames(plan_names, count.length, limit, offset, req.query.sort, req.query.system)
+
+        trace.leave();
+
+        return res.json(resp);
```

Comment:
Yeah, it seems odd but but with the `name` parameter we're just returning a list of remediation plan names and it's kind of unnecessary to continue with the rest of the function. And this is what's currently at the end of the function but it doesn't really make sense in for this case:

```
const resp = format.list(remediations, count.length, limit, offset, req.query.sort, req.query.system);
    return res.json(resp);
```

I think it would make sense to split this off into a separate function like you suggested when we do the refactoring that Rex mentioned.